From e1c9ffaaddb5e7d114cc6ed363e00df22c6fc344 Mon Sep 17 00:00:00 2001
From: fritsch <Peter.Fruehberger@gmail.com>
Date: Sun, 25 Jun 2017 18:40:59 +0200
Subject: [PATCH] VAAPI: Explicitely don't open for hevc 10 bit. Some distros
 package systemwide ffmpeg and their users en up with a black screen for that
 kind of videos. As kodi krypton is meant to be used with ffmpeg 3.1 this is
 not needed for us.

---
 xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp
index ed27cbbb6ec9..2f6502561de8 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp
@@ -558,6 +558,11 @@ bool CDecoder::Open(AVCodecContext* avctx, AVCodecContext* mainctx, const enum A
     }
     case AV_CODEC_ID_HEVC:
     {
+      // for distributions using far newer ffmpeg
+      // not available when Krypton was released
+      if (avctx->profile == FF_PROFILE_HEVC_MAIN_10)
+        return false;
+
       profile = VAProfileHEVCMain;
       if (!m_vaapiConfig.context->SupportsProfile(profile))
         return false;
